src/seamless-immutable.js

```diff
@@ -471,8 +471,17 @@
     return makeImmutable(obj, mutatingObjectMethods);
   }
 
+  var REACT_ELEMENT_TYPE = typeof Symbol === 'function' && Symbol.for && Symbol.for('react.element');
+  var REACT_ELEMENT_TYPE_FALLBACK = 0xeac7;
+
+  function isReactElement(obj) {
+    return obj.hasOwnProperty &&
+           obj.hasOwnProperty('$$typeof') &&
+           obj.$$typeof === REACT_ELEMENT_TYPE_FALLBACK || obj.$$typeof === REACT_ELEMENT_TYPE;
+  }
```

Comment:
I initially just copied that function but I had it fail in production in IE11 because Symbol is defined but not implemented. Could also be an issue with the polyfill I was using. 

It would be nice if it was the same implementation, but I couldn't get it to work like that. I need to write some tests for this. Do you think I should add a dev dependency on React so I can test the detection?

> On 13 Apr 2016, at 22:18, Alexey Shamrin <notifications@github.com> wrote:
> 
> In src/seamless-immutable.js:
> 
> > @@ -471,8 +471,17 @@
> >      return makeImmutable(obj, mutatingObjectMethods);
> >    }
> >  
> > +  var REACT_ELEMENT_TYPE = typeof Symbol === 'function' && Symbol.for && Symbol.for('react.element');
> > +  var REACT_ELEMENT_TYPE_FALLBACK = 0xeac7;
> > +
> > +  function isReactElement(obj) {
> > +    return obj.hasOwnProperty &&
> > +           obj.hasOwnProperty('$$typeof') &&
> > +           obj.$$typeof === REACT_ELEMENT_TYPE_FALLBACK || obj.$$typeof === REACT_ELEMENT_TYPE;
> > +  }
> What do you think about following React.isValidElement implementation more closely?
> 
> function isReactElement (obj) {
>   return (
>     typeof obj === 'object' &&
>     obj !== null &&
>     obj.$$typeof === REACT_ELEMENT_TYPE_FALLBACK || obj.$$typeof === REACT_ELEMENT_TYPE
>   );
> }
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>